pythran/pythonic/include/types/nditerator.hpp

```diff
@@ -12,15 +12,9 @@ namespace pythonic
      */
     template <class E>
     struct nditerator
-        : public std::iterator<
-              std::random_access_iterator_tag,
-              typename std::remove_reference<decltype(
-                  std::declval<E &>().fast(0))>::type,
-              ptrdiff_t, typename std::remove_reference<decltype(
-                             std::declval<E &>().fast(0))>::type *,
-              typename std::remove_reference<decltype(std::declval<E &>().fast(
-                  0))>::type /* no ref here, because we return a proxy object */
-              > {
+        : public std::iterator<std::random_access_iterator_tag,
+                               typename std::remove_reference<decltype(
+                                   std::declval<E &>().fast(0))>::type> {
```

Comment:
Which is a dangling reference isn't it? As the iexpr is generated from the fast operator.